builtin/providers/aws/resource_aws_nat_gateway.go

```diff
@@ -101,7 +101,7 @@ func resourceAwsNatGatewayRead(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
-	if ngRaw == nil || strings.ToLower(state) == "deleted" {
+	if ngRaw == nil || strings.ToLower(state) == "deleted" || strings.ToLower(state) == "deleting" || strings.ToLower(state) == "failed" {
```

Comment:
Would it be possible to re-factor this code a little? I know that you are just adding more checks, but it is getting pretty ugly. We could only do `strings.ToLower(*resp.NatGateways[0].State)` once, for example. I would also move to using a lookup table to check for many states, but that would be a personal preference.